modules/raft/src/main/java/org/apache/ignite/internal/raft/RaftGroupServiceImpl.java

```diff
@@ -452,9 +459,12 @@ public <R> CompletableFuture<R> run(Command cmd) {
         Function<Peer, ActionRequest> requestFactory;
 
         if (cmd instanceof WriteCommand) {
+            byte[] commandBytes = commandsMarshaller.marshall(cmd);
+
             requestFactory = targetPeer -> factory.writeActionRequest()
                     .groupId(groupId)
-                    .command((WriteCommand) cmd)
+                    .command(commandBytes)
+                    .deserializedCommand((WriteCommand) cmd)
```

Comment:
Please give more description here, why pass the command in raw form.

Reply:
Ok

Reply:
Maybe we can add more? for example, "so that the local node does not desiralize the command several times".